fix(rdb): update known-differences (#3378)
scaleway/docs-content: serverless/sql-databases/reference-content/known-differences.mdx

    PREPARE TRANSACTION 'transactionid';
    ```

> - Discarding all session state using `DISCARD ALL` command can be performed, but results are not guaranteed. Specifically, some session state variable might be kept if multiple SQL clients are performing operations concurrently.
> 
>     ```sql
>     DISCARD ALL;
>     ```
> 
> - `pg_advisory_lock` command can be used, but lock is not guaranteed. As a consequence, third party tools using this feature (such as storing [Terraform state using pg backend](https://developer.hashicorp.com/terraform/language/settings/backends/pg)) will not guarantee state locking if multiple Terraform clients are editing the same state concurrently. 
> 
>     ```sql
>     SELECT * FROM pg_advisory_lock(1);
>     ```
> 

Reviewer: ```suggestion
- Discarding all session states using the `DISCARD ALL` command is possible, but results are not guaranteed. Specifically, some session state variables might be kept if multiple SQL clients perform operations concurrently.

    ```sql
    DISCARD ALL;
    ```

- `pg_advisory_lock` command can be used, but the lock is not guaranteed. As a consequence, third-party tools that use this feature, such as Terraform when storing [Terraform state using pg backend](https://developer.hashicorp.com/terraform/language/settings/backends/pg). State locking is not guaranteed if multiple Terraform clients are editing the same state concurrently. 

    ```sql
    SELECT * FROM pg_advisory_lock(1);
    ```

```
